sdk/angelscript/projects/cmake/CMakeLists.txt

```diff
@@ -193,21 +193,30 @@ export(EXPORT AngelscriptTargets
 	FILE "${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptTargets.cmake"
 	NAMESPACE Angelscript::
 )
-configure_file(cmake/AngelscriptConfig.cmake
-	"${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptConfig.cmake"
-	COPYONLY
-)
+
+include(CMakePackageConfigHelpers)
 
 set(ConfigPackageLocation lib/cmake/Angelscript)
+
+configure_package_config_file(
+  "cmake/Config.cmake.in"
+  "AngelScriptConfig.cmake"
```

Comment:
You should always respect the out-of-source approach. Destination of `AngelScriptConfig.cmake` should be inside `CMAKE_CURRENT_BINARY_DIR`. Example:
- https://github.com/forexample/package-example/blob/00335ff3f39c3b82a4b4b6095412b5a9e3aeda81/Foo/CMakeLists.txt#L139-L171

Reply:
I was trying to change as few lines of original file as possible, it uses a bit different way to setup paths. Should I convert it to approach from above example?

Reply:
> I was trying to change as few lines of original file as possible

If you adding new lines the number of lines doesn't matter actually. For the conflicts the number of lines you're **changing** matters.

> Should I convert it to approach from above example?

Yes, you can use it as a template.

Reply:
Also just for the record, this part is not about hunterization, it's a general CMake improvement. They should be sent upstream, so we don't have to maintain them.
- https://docs.hunter.sh/en/latest/creating-new/patch.html

Reply:
@ruslo this boilerplate is just insane, I converted it but now the cmake/ dir in libdir is not even created and I can't debug it - messages are not printed, and I'm able to only achieve it under jenkins build so it requires a commit+push+sha1 calc roundtrip which is becoming a bit frustrating. Sadly CMake has no sanitizing or any sort of standardization so it either works or not, and will not error when something is wrong, but sometime later. I will try to look into it sometime later maybe.

Thanks for all the help!